lua_modules/http/httpserver.lua

```diff
@@ -23,54 +23,49 @@ do
   -- response methods
   ------------------------------------------------------------------------------
   local make_res = function(csend, cfini)
-   local send = function(self, data, status)
-    -- TODO: req.send should take care of response headers!
-    if self.send_header then
-      csend("HTTP/1.1 ")
-      csend(tostring(status or 200))
-      -- TODO: real HTTP status code/name table
-      csend(" OK\r\n")
-      -- we use chunked transfer encoding, to not deal with Content-Length:
-      --   response header
-      self:send_header("Transfer-Encoding", "chunked")
-      -- TODO: send standard response headers, such as Server:, Date:
+   -- send("hello world\r\n") will send the data
+    -- send() will close connection
+    local send = function(data)
+      if data then
+        -- chunked transfer encoding
+        csend(("%X\r\n"):format(#data))
+        csend(data)
+        csend("\r\n")
+      else
+        -- finalize chunked transfer encoding
+        csend("0\r\n\r\n")
+        -- close connection
+        cfini()
+      end
     end
-    if data then
-      -- NB: no headers allowed after response body started
-      if self.send_header then
-        self.send_header = nil
-        -- end response headers
+    local send_header = function(header)
+      if type(header) ~= "table" then
+        print("httpserver.lua: header need a table! eg: {status = 200, headers = {Connection = \"close\"}}")
+        return
+      end
+      if type(header.headers) ~= "table" then
+        print("httpserver.lua: header.headers need a table!")
+        return
+      end
+      -- status
+      local check = status_code[header.status]
+      if check == nil then check = 200 else check = header.status end
+      csend(("HTTP/1.1 %s\r\n"):format(status_code[check]))
```

Comment:
That's not terribly idiomatic Lua and defaulting to success is worrying.  I'd prefer, I think,
`csend(("HTTP/1.1 %s\r\n"):format(status_code[status] or status))`
which replaces `header.status` only if it's found in the table and otherwise passes exactly what's been given.  If you want the API to default to 200, which... well, maybe... then add a `status = status or 200` earlier in the function.

Reply:
like this ?
```lua
-- status maybe `404` or `Not Found`
function send_status(status)
    status = status or 200
    csend(("HTTP/1.1 %s\r\n"):format(status_code[status] or status))
end
```

Reply:
I think that's correct code, though I don't think that "Not Found" is an acceptable `status` value?  AFAICT the acceptable values are of the form `404` (the integer), `"404"` (the string), or `"404 Not Found"` (the string), yes?

Reply:
but ```status_code``` defines "404 not found", so this seems to be correct to me.

Reply:
@HHHartmann The comment above the function suggests that "Not Found" is an acceptable `status` value.  That, I think, is not true.  The comment should be something like
```
-- status may be any of: an int code (`404`), stringified int code (`"404"`), or code and text (`"404 Not Found"`)
```
(As an aside, English is a terrible language; "maybe" and "may be" mean very different things).

Reply:
Ok I agree. Didn't get that you meant the comment.
I hope that @hanfengcan will pick up on this PR. Would really be a pitty it his work would get lost.